apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Tests/Logging/GoogleLoggerScopeTest.cs

```diff
@@ -0,0 +1,101 @@
+            Assert.Null(GoogleLoggerScope.Current);
+            using (new GoogleLoggerScope("grandparent"))
+            {
+                Assert.Equal("grandparent => ", GoogleLoggerScope.Current.ToString());
+                using (new GoogleLoggerScope("parent"))
+                {
+                    Assert.Equal("grandparent => parent => ", GoogleLoggerScope.Current.ToString());
+                    using (new GoogleLoggerScope("child"))
+                    {
+                        Assert.Equal("grandparent => parent => child => ", GoogleLoggerScope.Current.ToString());
+                    }
+                }
+                Assert.Equal("grandparent => ", GoogleLoggerScope.Current.ToString());
+                using (new GoogleLoggerScope("other_parent"))
+                {
+                    Assert.Equal("grandparent => other_parent => ", GoogleLoggerScope.Current.ToString());
+                }
+                Assert.Equal("grandparent => ", GoogleLoggerScope.Current.ToString());
+            }
+            Assert.Null(GoogleLoggerScope.Current);
+        }
+
+        [Fact]
+        public async Task Multiple_Tasks()
+        {
+            Func<string, string, string, Task> func = async (grandparent, parent, child) =>
+            {
+                Assert.Null(GoogleLoggerScope.Current);
+                using (new GoogleLoggerScope(grandparent))
+                {
+                    Assert.Equal($"{ grandparent} => ", GoogleLoggerScope.Current.ToString());
+                    await Task.Yield();
+                    using (new GoogleLoggerScope(parent))
+                    {
+                        Assert.Equal($"{grandparent} => {parent} => ", GoogleLoggerScope.Current.ToString());
+                        await Task.Yield();
+                        using (new GoogleLoggerScope(child))
+                        {
+                            Assert.Equal($"{grandparent} => {parent} => {child} => ", GoogleLoggerScope.Current.ToString());
+                            await Task.Yield();
+                        }
+                        await Task.Yield();
+                        Assert.Equal($"{grandparent} => {parent} => ", GoogleLoggerScope.Current.ToString());
+                    }
+                    await Task.Yield();
+                    Assert.Equal($"{ grandparent} => ", GoogleLoggerScope.Current.ToString());
+                }
+                Assert.Null(GoogleLoggerScope.Current);
+            };
+
+            await Task.WhenAll(
+                    Task.Run(() => func("grandparent-one", "parent-one", "child-one")),
+                    Task.Run(() => func("grandparent-two", "parent-two", "child-two")),
+                    Task.Run(() => func("grandparent-three", "parent-three", "child-three")),
+                    Task.Run(() => func("grandparent-four", "parent-four", "child-four")),
+                    Task.Run(() => func("grandparent-five", "parent-five", "child-five")));
+        }
+    }
```

Comment:
Yes, what you've added to this test is good because it now tests across two execution contexts, but you may also want to consider some additional tests similar to what I added in `SimpleManagedTracerTest`. What you have here (and my example above) are when the child execution context begins and ends all while the root scope is open. That is what I tested in `MultipleSpans_Async` and `MultipleSpans_Threads_Running_During_Span`.

Additional scenarios to consider testing with examples from `SimpleManagedTracerTest`:
* `MultipleSpans_Threads_Started_Before_Span` - child execution context starts _before_ the root scope is open, but child scopes within that context are then started _while_ the root scope is open. In this case they _should not_ be owned by the root span.
* `MultipleSpans_Threads_Started_During_Span` - child execution context starts _while_ the root scope is open, but child scopes within that context are then started _after_ the root scope is closed. In this case they _should_ be owned by the root span. 

I'm not sure if you have the same "should/should not"s here, but you may want to add similar tests which assert for your expectations.